src/main/resources/static/js/app.js

```diff
@@ -0,0 +1,119 @@
+var app = angular.module("JournalApp", []);
+
+app.controller("CategoryController", function ($scope, $http) {
+
+    $scope.isAnySubscription = function() {
+        var result = false;
+        for(var i=0; i< $scope.subscriptions.length; i++) {
```

Comment:
The java code is nicely written using Java 8 stream, but the JavaScript code is quite imperative. For example, this function can be turned into something like:

```javascript
$scope.isAnySubscription = () => $scope.subscriptions.reduce((v1, v2) => v1 || v2).active;
```

Concise and clear code. Less bugs later.